src/mlpack/core/tree/rectangle_tree/r_plus_tree_descent_heuristic_impl.hpp

```diff
@@ -43,8 +43,6 @@ size_t RPlusTreeDescentHeuristic::ChooseDescentNode(TreeType* node,
         node->Child(bestIndex).Bound();
     bound |= node->Dataset().col(point);
 
-    success = true;
-
```

Comment:
The change that you've made means that during this loop, `success` is not reinitialized to `true` each iteration.  So this introduces a bug.